app/controllers/general.php

```diff
@@ -300,6 +301,35 @@
 
                 Authorization::setRole('role:' . Auth::USER_ROLE_APP);
                 Authorization::setDefaultStatus(false);  // Cancel security segmentation for API keys.
+
+                if (time() > ($key->getAttribute('accessedAt', 0) + (24 * 60 * 60))) {
+                    $key->setAttribute('accessedAt', time());
+                    $sdkValidator = new WhiteList([
+                        'appwrite:nodejs',
+                        'appwrite:deno',
+                        'appwrite:php',
+                        'appwrite:python',
+                        'appwrite:ruby',
+                        'appwrite:go',
+                        'appwrite:java',
+                        'appwrite:dotnet',
+                        'appwrite:dart',
+                        'appwrite:kotlin',
+                        'appwrite:swift'
+                    ], true);
+    
+                    $sdk = $request->getHeader('x-sdk-version', 'UNKNOWN');
```

Comment:
I think we might be able to add a set of specific headers to the sdks like
* x-sdk-name
* x-sdk-version

Which will make parsing easier. We can use the full combined string for the standard `user-agent` header. WDYT?

Reply:
yes! that would remove all the parsing logic.